packages/nvidia_nat_core/tests/nat/data_models/test_optimizable.py

```
@@ -358,3 +358,26 @@ class MyModel(OptimizableMixin):
+
+    @pytest.mark.parametrize("fmt", ["f-string", "jinja2", "mustache"])
+    def test_accepts_valid_formats(self, fmt):
+        s = SearchSpace(is_prompt=True, prompt="test", prompt_format=fmt)
+        assert s.prompt_format == fmt
+
+    def test_accepts_none(self):
+        s = SearchSpace(is_prompt=True, prompt="test", prompt_format=None)
+        assert s.prompt_format is None
+
+    def test_rejects_invalid_format(self):
+        with pytest.raises(Exception):  # Pydantic ValidationError
```

Ruff (0.15.0)

[warning] Do not assert blind exception: `Exception`

(B017)